python/src/llvm.cc

```diff
@@ -377,6 +379,26 @@ void init_triton_llvm(py::module &&m) {
               fpm.addPass(BreakStructPhiNodesPass());
               fpm.addPass(InstCombinePass());
             });
+        bool enableAddressSanitizer =
+            mlir::triton::tools::getBoolEnv("TRITON_ENABLE_ADDRESS_SANITIZER");
+        if (enableAddressSanitizer) {
+          llvm::SmallVector<StringRef, 3> split;
+          StringRef(mod->getTargetTriple()).split(split, '-');
+          if (split[0] != "amdgcn") {
+            std::string ErrMsg =
+                "Address Sanitizer Error: Address sanitizer is currently only "
+                "supported on the AMD backend\n";
+            throw std::runtime_error(ErrMsg);
+          }
+          for (llvm::Function &f : mod->functions()) {
```

Comment:
This can happen in AMD backend as preprocessing?

Reply:
Updated